### ISSUE TITLE
Android Mobile App Intermittently Slow

### ISSUE THREAD

**Issue by MaxK99** [home-assistant/android#3396]
**Home Assistant Android app version(s):**
2023.1.1-full
**Android version(s):**
13
**Device model(s):**
pixel 4a
**Home Assistant version:**
2023.2.5
**Last working Home Assistant release (if known):**

**Description of problem, include YAML if issue is related to notifications:**
The Android mobile app is intermittently slow to startup (5-10 seconds to start) and slow when navigating in the app (up to 5 seconds to navigate to a different screen). The issue is persistent when it occurs until the phone is rebooted. This issue has been reported by others in the Home Assistant Forum

https://community.home-assistant.io/t/mobile-app-2022-11-0-full-slow-lately-solved-after-reset-but-now-slow-again/504816

https://community.home-assistant.io/t/ha-app-has-become-slow-on-android/534341/6

**Companion App Logs:**
[homeassistant_companion_log_1-28-2023_5-59-6.txt](https://github.com/home-assistant/android/files/10887361/homeassistant_companion_log_1-28-2023_5-59-6.txt)

```

```

**Screenshot or video of problem:**

**Additional information:**

**Comment by jpelgrom** [comment]
I encountered something similar a few times, looked like it was something in the WebView causing it to slow down as other parts of the app remained fast.

Can you check:
- when the app is slow, do other features like device controls, notifications, widgets get slow?
- does force stopping the app fix the issue?

**Comment by MaxK99** [comment]
Unfortunately, it can be a few weeks before the issue occurs. But I will do those checks when it does happen again.  Hopefully, others will jump in and provide feedback sooner.

There was a recent [mention](https://community.home-assistant.io/t/ha-app-has-become-slow-on-android/534341/14) about how clearing the cache on the phone resolved the same issue.

**Comment by g-kiss** [comment]
This issue occurs for me too. The cache clearing does not solve the problem, only the mobile restart.

**Comment by ciB89** [comment]
I also have this issue. I often also get the screen that the instance is not available (where I can decide to wait, check internal URL etc.) when I open up the app. It's like there is a massive delay/package loss.

**Comment by Mincka** [comment]
In my case, I usually "fix" this by rebooting the phone but it comes back quickly. I am using a Google Pixel Pro 7 but I remember I also had this behavior with my Galaxy S20+ few months back, this is clearly not something recent.
I just tried to purge the cache but it did not change anything, however, **I used "Force Stop" and it immediately fixed it**.
And I mean not just swiping up the app in the app switcher, it does not stop really stop it.

**Comment by dshokouhi** [comment]
Based on this [comment](https://github.com/home-assistant/android/issues/31#issuecomment-555794331) and the description from this issue I think this might be a frontend issue and not an app issue.

`Home Assistant should automatically reconnect with a maximum of 1 second delay.`



> I often also get the screen that the instance is not available (where I can decide to wait, check internal URL etc.) when I open up the app.

This is unrelated to this issue. The only reason why that pop-up will show up is either a connection error to the server or your server did not respond within 10 seconds.



> And I mean not just swiping up the app in the app switcher, it does not stop really stop it.

Some manufacturers will implement force stopping an app when you swipe it away but typically that does not happen and it just stops the activity.

**Comment by ciB89** [comment]
> > I often also get the screen that the instance is not available (where I can decide to wait, check internal URL etc.) when I open up the app.
> 
> This is unrelated to this issue. The only reason why that pop-up will show up is either a connection error to the server or your server did not respond within 10 seconds.


Well, I only get this message when I open up the app, and webview works perfectly fine and responsive, so I guess this seems to be at least not a network issue.

**Comment by Mincka** [comment]
I can confirm the behavior described by @ciB89. It becomes so slow at some point that the message is shown.
The slowness is the same on Wi-Fi or 4G, using internal or external URL.
It's like having a throttling preset to Slow 3G enabled.

**Comment by jpelgrom** [comment]
I actually have a video of it being slow to load from last December with dev tools open, showing unexpected gaps in the timeline with the webview waiting on basically nothing? Not sure where to start with determining where the issue lies / how to fix it though, and I haven't seen this behavior recently.


https://user-images.githubusercontent.com/8148535/223224266-47917152-fe71-4d3a-883d-4e3e8d667b2a.mp4

**Comment by Mincka** [comment]
Maybe it's not related to network but to high CPU usage due to multiple instances of the app (as seen in #2749 for instance, no pun intended).
If one clicks on notifications on a regular basis, new instances are spawn and the overall allocated CPU becomes high. Consequently, it degrades slowly over time.
This could be the reason why killing the app fixes the issues, all the instances are killed.

**Comment by dshokouhi** [comment]
> Maybe it's not related to network but to high CPU usage due to multiple instances of the app (as seen in #2749 for instance, no pun intended).
> If one clicks on notifications on a regular basis, new instances are spawn and the overall allocated CPU becomes high. Consequently, it degrades slowly over time.
> This could be the reason why killing the app (or rebooting the phone) fixes the issues, all the instances are killed.

That code has been there for a very long time, as in well over a year. This issue has only recently started back in November so the timing and dates don't match up.

**Comment by liollury** [comment]
Hello, I also have this problem and it started at the same time as adding an Android widget and quick access with the notification bar (maybe a coincidence) . I notice too this is temporarely resolve with device reboot.

**Comment by Mincka** [comment]
It's not easy to reproduce but I ran a top and saw this while the application was slower to load the home page.
![image](https://user-images.githubusercontent.com/6184289/223667942-96d537c8-8909-4e64-b5e4-6d8296d57c8e.png)

Sorry the cropped output... PID 11884 is the webview:
![image](https://user-images.githubusercontent.com/6184289/223669291-a75aaf7a-a4e7-4dad-94e6-73358551483d.png)

**Comment by dshokouhi** [comment]
@Mincka managing CPU tasks is out of scope for any android app, that is up to the system to handle it. 

Also we no longer keep those newly spanned tasks in history, they are removed as soon as they user navigates away from it. That change was also done a while ago.

If that were indeed the issue then it would be very easy to reproduce actually.

**Comment by Mincka** [comment]
Fine, but if you have some helpful recommendations to find out the root cause, don't hesitate to share.

**Comment by dshokouhi** [comment]
At this point its probably best we look for whats common amongst all the users.

1. What version of HA core the user is on (even better if we can tell which version the issue started as we suspect this started in November 2022)
2. What version of the app the user is on
3. What WebView package and version is installed on the device
4. If the user has any custom cards or components (may be trivial as I think some users don't have anything custom). For this keep in mind we care about if the card is loaded as a resource in Lovelace, it does not have to be used just loaded.
5. What type of HA setup, HA core, Docker, Supervisor, HA OS
6. Anything else that we may consider to be common.

For me I am unable to reproduce this on my Pixel 7 Pro. HA core is on 2023.3 (only running HA core), WebView package is Googles and version is 110. Only custom card I use is custom weather and auto entities.

**Comment by MaxK99** [comment]
@dshokouhi Edited original post with requested details.  I have not had a recurrence since original post yet.

**Comment by SwannyBFC** [comment]
Also see this exact problem on pixel 6 pro with the latest version of the companion app. Loading HA  via a browser on the same phone works quick.

**Comment by wouterrutgers** [comment]
Hi,

Having this problem too since a few months.

What version of HA core the user is on (even better if we can tell which version the issue started as we suspect this started in November 2022)
2023.3.4

What version of the app the user is on
2023.1.1-full

What WebView package and version is installed on the device
110.0.5481.153

If the user has any custom cards or components (may be trivial as I think some users don't have anything custom). For this keep in mind we care about if the card is loaded as a resource in Lovelace, it does not have to be used just loaded.
-> WebRTC camera
-> Frigate
-> Powercalc
-> Circadian Lighting
-> Pyscript
-> Dahua
-> Fontawesome
-> Dwains Dashboard
-> Volkswagen We Connect ID
-> Daikin Residential Controller

What type of HA setup, HA core, Docker, Supervisor, HA OS
HA OS

Anything else that we may consider to be common.
I have this bug on a Samsung Galaxy S22 Ultra.

**Comment by felipecrs** [comment]
With this bug the companion app is basically unusable. It takes 50 seconds to load on the app versus 5 seconds on Chrome.

https://user-images.githubusercontent.com/29582865/225815334-fa303024-0fc3-4e65-9e06-05036b8e4f55.mp4

**Comment by bmcclure** [comment]
I've also been experiencing this issue for a month or two at least. I assumed I was having a WiFi problem until I tested the web app and it loaded really quickly.

Every time I try and open the Android app it spins for a while, gives me the error about not finding my instance, then starts loading a few seconds later.

But then every page I go to within HA just sits on a blank window for 5-10 seconds until it starts loading in slowly.

**Comment by MaxK99** [comment]
I am experiencing this issue again since first posting the details above.  Nothing has changed version-wise except I have the March 5, 2023 security update.  

There is nothing in the Core or Supervisor logs.  I did captured the companion log again.
[homeassistant_companion_log_2-18-2023_14-52-4.txt](https://github.com/home-assistant/android/files/11009345/homeassistant_companion_log_2-18-2023_14-52-4.txt)

@dshokouhi I will try to keep my phone in this state in case you want me to check anything else.

**Comment by OverFlow636** [comment]
Also experiencing this issue for the last 2months or so, it even affects the ble transmitter, so my espresense setup thinks I'm not home because the app stops broadcasting. I use the persistent websocket connection if that changes anything.

Notifications are also significantly delayed during this time, I have some door open notifications which will trigger 5 mins after sent instead of <1 second

When I open the app it's very slow to open, complains about ha server being unreachable then eventually starts up, but navigating between dashboards is very slow.

I just force stop the app and re open and it's fast again.

**Comment by bunicb** [comment]
Same here on a Pixel 6A and OnePlus Nord.
Versions of the companion app are basically always latest, Core HA is currently 2023.3.6.

Notifications are delayed at that time, location of apps in zones doesn't update correctly for a while.

Force stop seems to alleviate the problem for a while.

**Comment by vrmtm** [comment]
When the application starts running slowly, I looked at what it shows in debug mode. It seems that in addition to the main instance, there are also several others that are marked as detached. 
Killing/reloading the application not only does not help, but after each kill/reload a new detached instance appears.
The only thing that helps is rebooting the device. This helps for a few days or a week, then everything happens again.
![ha_error](https://user-images.githubusercontent.com/5072820/230736778-1a574e98-ed32-46c1-8ab7-3e79789f0f38.png)

**Comment by wouterrutgers** [comment]
To add to this, it seems that it is also draining my battery quicker.

I had removed the app for a while and used the web app, and my phone used to live a lot longer. Reinstalled the app because I missed the notifications (html5 notifications are not as reliable), and now it is getting drained a lot faster again.

No extra sensors or w/e enabled, just basic install with default settings. So might be related to this issue and the higher CPU usage others have noticed.

**Comment by dshokouhi** [comment]
@wouterrutgers for battery drain issues please refer to our troubleshooting guide, as far as I know no other user in this thread has mentioned high battery drain so its possible you have some settings to adjust or something that needs to be cleaned up.

https://companion.home-assistant.io/docs/troubleshooting/faqs#android-app-battery-drain

**Comment by VNRARA** [comment]
Try to remove the internal ip and wifi ssid from the companion app completely. I found that mine significantly sped up after doing that.

**Comment by VNRARA** [comment]
Can you try filling it in and restarting and removing it again?

**Comment by felipecrs** [comment]
Ok, just did. Let's see how it behaves.

**Comment by felipecrs** [comment]
Just to update, I tested it, and it has apparently helped for a while. Just like cleaning the app cache does.

But the problem returns.

**Comment by nagyrobi** [comment]
Same for me, unfortunately.

**Comment by bartgrefte** [comment]
Same here since a couple of days (could be a bit longer, don't use the app every day), though strangely it resolved itself sometime last night, without doing anything.

**What version of HA core the user is on (even better if we can tell which version the issue started as we suspect this started in November 2022)**
Started while on HASS OS 9.5 and Core 2023.4.4, but not directly after updating to these versions. The issue continued after updating to 10.1 / 2023.5.2.

**What version of the app the user is on**
2023.3.0-full

**What WebView package and version is installed on the device**
Android System WebView 112.0.5615.135.

**If the user has any custom cards or components (may be trivial as I think some users don't have anything custom). For this keep in mind we care about if the card is loaded as a resource in Lovelace, it does not have to be used just loaded.**
Nothing custom, as for addons: 
File editor, Glances, Mosquitto Broker, SQLite Web, Terminal & SSH.


**What type of HA setup, HA core, Docker, Supervisor, HA OS**
HA OS, OVA image in VMware Workstation 15.5.6, host OS W10 x64 LTSC 1809 (17763.1999).

**Anything else that we may consider to be common.**
Samsung Galaxy S20 FE 5G, DNS lookup is done using PiHole. Since the memory card of my RPi recently died and had to do a clean install, I thought the problem was there, seems it's not.

**Comment by clau-bucur** [comment]
Have the same issue, for quite a long time now.
Apparently clearing cache is the only short-term solution for the problem.

**Comment by energywave** [comment]
I confirm the same behavior from long time too. When it happens it can be resolved by clearing the app cache and force the app to exit, then it works again as normal, until it happens again.
Not the worse issue when you understand how to overcome but I'm sure there's many person that don't know what to do and uninstall/reinstall to resolve, losing all sensor configuration...

**Comment by werkstrom** [comment]
https://github.com/home-assistant/core/issues/100982 Confirm same behavior. More info in the linked issue

**Comment by smarthomefamilyverrips** [comment]
Same here!

**Comment by gitlaman** [comment]
Mine is set to auto clear cache every night with 1Tap Cleaner Pro, since i started the auto clear cache i disabled the auto reboot HA automation (i thought that the database was making the lags) and for 10 days the HA app is working flawless.

**Comment by smarthomefamilyverrips** [comment]
Yes clearing cache or reboot phone solves the problem for a while but this should not be the solution as before not had this issue

**Comment by earion68** [comment]
I had a similar issue lately and spent days investigating. 
Long story short, it was due to the file known_devices.yaml that was too big, like several megabytes (this was due to a chatty BLE device tracker integration).
Reduced the file to like ~1000 lines and after a HA reboot it was day and night.

**Comment by felipecrs** [comment]
Where can I find this file?

**Comment by earion68** [comment]
> Where can I find this file?

At the root of /config next to configuration.yaml

**Comment by nagyrobi** [comment]
@earion68 that's just a coincidence. No use of that file here (not even present), issue still appears occasionally.

**Comment by werkstrom** [comment]
Would have to initially side with @nagyrobi. on this one. A. That file is not present on my system and B. Restarting the phone would not (unless there is some specific code doing it) adjust that file on the server and reload the server (as my issues are resolved without server restart). Not saying your issue wasn't resolved as per your description @earion68 . Just saying it is likely another issue. IMHO. Thanks for your input though 👍

**Comment by adamantivm** [comment]
Same behavior for me since a few HA core releases ago. Interested in helping testing or doing anything I can to help nail down the issue.

**Comment by adamantivm** [comment]
> What version of HA core the user is on
Home Assistant 2023.10.3
Supervisor 2023.10.0
Operating System 11.0
Frontend 20230608.0 - latest

> What version of the app the user is on
2023.10.2-full

> What WebView package and version is installed on the device
117.0.5938.153

> If the user has any custom cards or components (may be trivial as I think some users don't have anything custom). For this keep in mind we care about if the card is loaded as a resource in Lovelace, it does not have to be used just loaded.
Hmm ... nothing that I wrote myself. The most advanced I have is a Thermostat card, but it's the defualt one as far as I know.

> What type of HA setup, HA core, Docker, Supervisor, HA OS
HA OS

> Anything else that we may consider to be common.
Works fast on Web but get slow on mobile.
Restarting Home Assistant core makes it fast again for a while, but after hours / a day it gets slow again.
Force stop on Android also makes it fast again - TBD for how long.

**Comment by werkstrom** [comment]
After some more testing I can confirm the issue is temporarily resolved by flushing the app cache. It does however eventually return. Very random when it occurs. I cannot see any pattern to it.

**Comment by Mincka** [comment]
Does anyone has this behavior but don't use or almost don't use notifications (clicked or not)?

It's just to check if the performance could decrease with notifications.

**Comment by nagyrobi** [comment]
I do use notifications...

**Comment by smarthomefamilyverrips** [comment]
I get 2 notification a day at 8:30 in morning and one in the night when Alarmo arms on a regular day

**Comment by werkstrom** [comment]
> Does anyone has this behavior but don't use or almost don't use notifications (clicked or not)?
> 
> It's just to check if the performance could decrease with notifications.

I use notifications, but the issues are very intermittent, i.e. can go several weeks with no issue. And I have several notifications a day sent from HA so correlation (in my case) between notifications and issues is very weak IMHO

**Comment by werkstrom** [comment]
It would be usefull with some sort of troubleshooting function. If it was possible to export the cash some how that would perhaps be useful. I don't know that exporting it is possible though ;(

**Comment by TomK** [comment]
I also use notifications, i send images from my doorbell when someone is present.

**Comment by Mincka** [comment]
Sorry guys, maybe I was not clear. I think it would be interesting to know if some people **don't use notifications but also have this problem**, so we could exclude that the issue is related to notifications.

**Comment by TomK** [comment]
> Sorry guys, maybe I was not clear. I think it would be interesting to know if some people **don't use notifications but also have this problem**, so we could exclude that the issue is related to notifications.

We need to establish a baseline.  if nobody replies saying they do not use notifications, we don't know if that's because everybody uses them or nobody does, or simply nobody replied.  Really we need a matrix table for this issue, people tally what features they use and what they don't.  Google forms?

**Comment by earion68** [comment]
> It would be usefull with some sort of troubleshooting function. If it was possible to export the cash some how that would perhaps be useful. I don't know that exporting it is possible though ;(

Definitely, this is what we need. Pretty good logging system on the backend, but lovelace is definitely missing one

**Comment by JurajNyiri** [comment]
I am experiencing this issue consistently. Clearing cache or killing the app did not help. I see a very slow start with a msg server is not available, which dissapears after a few seconds. Then it slowly loads, going to settings and for example devices takes a long time with nothing on the screen.

I installed the web version from chrome and that works instantly and without any issues.

Pixel 6 latest ha, app.

**Comment by werkstrom** [comment]
In my case I notice that only clearing the cache does help in the (very) short term. The app becomes responsive immediatly. However. The next time I open it up (even if only a few minutes later) it is back to the slow response times again. However, first clearing the cache, and then force-killing the app appears to help for a longer period of time.

I'd also like to note that when opening grafana within the app (from the left hand menu) the app is still very, very slow, and since my understaning is that grafana in this instance is running on it's own server (it's own docker image) I think it would be safe to say the issue lies within the app and not on the server. IMHO.

**Comment by CommanderROR9** [comment]
I have been seeing this issue for many months now. The app slows down for no discernible reason. WebUI is fine and app recovers after closing it and reloading it. Seems like some kind of memory leak or something like that, but if it were something obvious it would have been patched long ago.

**Comment by whosmatt** [comment]
Same behaviour on Pixel 8 Pro, temporarily fixed by force closing app.  
The reviews on google play make it seem like this is pretty common.

**Comment by werkstrom** [comment]
Can add this happens even when on a completely different WiFi (spending a few day at parents), connecting to HA via WireGuard VPN. Still on Pixel 7a. Force close end flush cache resolves issue immediatly

**Comment by ahlstromanton** [comment]
Still an issue on 2023.11.1, Pixel 7 Pro. 

While apparent when opening the app, the biggest issue this is causing for me is that notifications could be severely delayed. I've moved over to Telegram notifications in the meantime as this is unacceptable for my usecases.

**Comment by nagyrobi** [comment]
A quick note that since the latest HA update (2023.11) this is much frequent now. Force close+cache clear helps.

**Comment by gitlaman** [comment]
I still dont have problems after ~month i use auto clear cache every day only for HomeAssistant, i use Sd Maid SE for the auto cleaning on Android 13.

**Comment by Mincka** [comment]
> I still dont have problems after ~month i use auto clear cache every day only for HomeAssistant, i use Sd Maid SE for the auto cleaning on Android 13.

How can you say that "you don't have problems" when you need to pay for an app to clean the cache daily?
By the way, the last time you posted here, you were advertising another paid application. Are you here for spam?

**Comment by ahlstromanton** [comment]
> A quick note that since the latest HA update (2023.11) this is much frequent now. Force close+cache clear helps.

Same here, didn't have this issue before 2023.11 (though I'm not a heavy user of the companion app). Now, notifications are severely delayed, causing huge problems for time sensitive notifications (fire alarms, house alarms, medication reminiders etc).

**Comment by gitlaman** [comment]
> > I still dont have problems after ~month i use auto clear cache every day only for HomeAssistant, i use Sd Maid SE for the auto cleaning on Android 13.
> 
> How can you say that "you don't have problems" when you need to pay for an app to clean the cache daily? By the way, the last time you posted here, you were advertising another paid application. Are you here for spam?

I'm advertising?? No man im saying what helped with my issue, i know that the app have problems bht what can i do??? Do you have fix maybe so i can use Homeassistant without delays??? No you dont! And also i changed that crap app because it was not working as it should. As for Sd Maid SE you have trial so you can try, and 1$ for a year is not much so i can live as i should in my home for now until there is a real fix for the app. And no i am not advertising. You are free to use Homeassistant with lags.

**Comment by jpelgrom** [comment]
> Same here, didn't have this issue before 2023.11 (though I'm not a heavy user of the companion app). Now, notifications are severely delayed, causing huge problems for time sensitive notifications (fire alarms, house alarms, medication reminiders etc).

This issue concerns slowdown of the dashboard / WebView interface, not native features. If you're having problems with push notifications, please create a new issue.

**Comment by ahlstromanton** [comment]
> > Same here, didn't have this issue before 2023.11 (though I'm not a heavy user of the companion app). Now, notifications are severely delayed, causing huge problems for time sensitive notifications (fire alarms, house alarms, medication reminiders etc).
> 
> This issue concerns slowdown of the dashboard / WebView interface, not native features. If you're having problems with push notifications, please create a new issue.

It is also slow to do the initial request (connecting to server), before loading any dashboards. 
You closed my other isse as a duplicate (#4001).

**Comment by nagyrobi** [comment]
> It is also slow to do the initial request (connecting to server), before loading any dashboards.

Confirming! It even times out and shows a popup with the option to change the URL or wait/retry.
This doing on local network, or remotely.

**Comment by MarshallCS** [comment]
This is occurring to me as well. Doing a force quit on the app resolves the issue temporarily

**Comment by jeremykeen** [comment]
Severe slowdown in the companion app makes it unusable for me. I have 2 HA instances, one local (2023.11.1) and one remote (2023.10.5), both having the same issue in the companion app. Web is fast. This just started happening in the last week or so.

Pixel 7 Pro
App version 2023.10.2

I have not tried force closing or clearing cache. Is there anything I can grab to help troubleshoot?

**Comment by Mincka** [comment]
>  I do have BLE transmitter enabled.
I don't, so unrelated for that one.

**Comment by rsaffi** [comment]
I _think_ it _could_ be related to the sensors that the companion can expose to HA? I haven't done any actual deep dive to investigate, but I think that it's a potential source of problem (say, memory leak?). 
One way to (cumbersomely, I admit) mitigate could be to disable all and start to enable them back one by one, with enough spacing between those to see whether the issue comes back or not. The issue is that sometimes (at least for me) it takes 2 weeks until the slowness reappears, so debugging this way would take a long time.

**Comment by clau-bucur** [comment]
> Sorry guys, maybe I was not clear. I think it would be interesting to know if some people **don't use notifications but also have this problem**, so we could exclude that the issue is related to notifications.

I was "suffering" too of this problem, very often. My system was relying heavily on notifications to inform me of different things.

After seeing your message here, I trimmed down many of my notifications (sent them to log files instead) and created automations to clear existing notifications in a few hours after they are posted.

After doing this, I have NEVER had the issue described here again. So I can say that in my case it seems 100% related to notification.

**Comment by nagyrobi** [comment]
I do use notifications, getting only 2-3 a week. And the slowness is almost daily.

**Comment by Benniepie** [comment]
Using Samsung Z Fold 5 / Android 13 and barely using the home assistant app notifications (use telegram instead). I have got into the routine of closing/reopening the app multiple times a day. It slows to a crawl particularly on dashboards with auto-entities or when using entity/icon drop downs initially but eventually it lags to the point the whole thing becomes unusable, I assumed the number of entitles in my HA had increased to the point my setup couldn't handle it (there are so many I should disable that I never use) until I found this thread - no idea if this is the same issue, but here's my setup+error from when it occurred if it helps:

HA Core: 2023.11.2
App version: 2023.11.2-full

Possible related error:
Logger: homeassistant.components.websocket_api.http.connection
Source: components/websocket_api/http.py:203
Integration: Home Assistant WebSocket API 
First occurred: 02:30:13 (4 occurrences)
Last logged: 02:30:56

[XXXXXXXXXX] Bennie Pie from 10.0.XX.XX (Mozilla/5.0 (Linux; Android 13; SM-F946B Build/TP1A.220624.014; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/119.0.6045.66 Mobile Safari/537.36 Home Assistant/2023.11.2-11694 (Android 13; SM-F946B)): Client unable to keep up with pending messages. Reached 4096 pending messages. The system's load is too high or an integration is misbehaving; Last message was: {"id":2227,"type":"result","success":true,"result":null}

**Comment by maurerle** [comment]
I also have this issue that the mobile app (play store or f-droid) are very slow.
I am using the mobile page as a workaround which is considerable better.

**Comment by bachoo786** [comment]
I
[homeassistant_companion_log_0-15-2024_15-19-45.txt](https://github.com/home-assistant/android/files/13940492/homeassistant_companion_log_0-15-2024_15-19-45.txt)
 am having the same issue as described in the OP. "Force stop" the HA app and it works fine. I have attached my app logs.

**Comment by vbueschken** [comment]
At the moment the app starts slower and slower every day...
Every 3-4 days I have to stop the app and clear the cache to have an acceptable performance

**Comment by Saduff** [comment]
> At the moment the app starts slower and slower every day...
> Every 3-4 days I have to stop the app and clear the cache to have an acceptable performance

I just use the PWA (open in browser -> Install/Add to home screen), no slowness issues with that. The Android app is only for push notifications.

**Comment by energywave** [comment]
> The Android app is only for push notifications.

That's not true at all! Benefits of using the app is the switching from internal/external address, all the sensors that the app sends to Home Assistant (I use them a lot! Think about the location, the zone you're in, SSID connected, if you're sleeping, etc etc), the reading of NFC, the integration with android smartwatches, toggles in the android pane of toggles, configurable widgets (!) and yes, push notifications that are a very powerful feature of the app.
So maybe it's the same for your specific use case but that's not true for all!

**Comment by Saduff** [comment]
Yeah, I mostly meant that I only use the Android app for push notifications.

> all the sensors that the app sends to Home Assistant

Wouldn't those still work with the app in the background? The slowness is only when opening the app.

I also use the Android smartwatch integration, but that works fine without ever opening the Android app on the phone.

So it's more like Android app for background tasks and PWA if you actually want to open HA.

**Comment by energywave** [comment]
> Yeah, I mostly meant that I only use the Android app for push notifications.

Oh ok, I interpreted your sentence in another way, sorry if I was too rude... :)

> > all the sensors that the app sends to Home Assistant
> Wouldn't those still work with the app in the background? The slowness is only when opening the app.

That's an interesting question! It could be true... Some tests must be done when the app is in that condition about sensors.

> I also use the Android smartwatch integration, but that works fine without ever opening the Android app on the phone.

Ok, not opening but at least you have to have the app running in background (I suppose, but maybe the app on the sw can contact autonomosly HA...? However that's OT)

> So it's more like Android app for background tasks and PWA if you actually want to open HA.

That could be an interesting workaround in the meanwhile that this long running problem will be fixed (finger crossed...)
That said the correct solution is that the app should work as it's expected, let's hope that it will be possible to bust this bug sooner or later.

**Comment by eric10k93** [comment]
This is interesting, I expected this to be a new problem, as I am now experiencing this as well.  It seems like something specific must've changed recently in how the app loads, as I had never had a slowdown problem until 2024.1.1 (and later)

**Comment by contactnat** [comment]
yes, this started for me somewhat recently too, although I can't pin point when. I keep HA up to date along with add ons and HACs, and I keep the android app up to date. 
I tried the clear cache in the android app work around which seemed to work, It slowed down within a few hours again though, so I tried another method I found (maybe on Reddit) that suggested to delete the known_devices.yaml file and restart HA - since then (now going on approximately 16hrs) its been good. waiting for the slow down to return though.

**Comment by contactnat** [comment]
> @contactnat - where can I find the known_devices.yaml file ?

In your homeassistant, find it in the same location as configuration.yaml using either the file editor addon or visual studio addon.
I also read somewhere that It may have been deprecated already so if you are new to ha it may not be there - I'm not sure, so have a look

**Comment by vbueschken** [comment]
@contactnat - I'm new (1.5 years) with HA- thats the reason I couldn't find the known_devices.yaml file.
So there must be another reason for the app becomes slower and slower every day.

**Comment by contactnat** [comment]
@vbueschken yeah I don't know how or why that file would be the cause, especially if it has been deprecated, my app is working fast now for almost 3 days and all I have done is the cache clear (on Android app) phone restart, ha restart, and the file delete then ha restart as above. Fingers crossed mine stays good, hope you get yours resolved - it's very frustrating for sure.

**Comment by alexrec-alt** [comment]
Was using outdated app from last year. After upgrading the server I decided to update Android app and nightmare began.

At app startup splash screen blinking few times, takes long to get into the working app. After that is says no connection or something but resolves in some time. Occassionally loosing connection when navigating.

What helped: just cleared App Cache. After that it start as normal heavy app. It starts with no blinking, much faster and no connection issues anymore.

Using the browser still faster tho.

**Comment by maurerle** [comment]
I had the problem, that (probably through privacy extension and changing ipv6 addresses) the list of IP addresses from the FritzBox integration created about 10k sensors over time. This did slow down the mobile app as well as web access.
I removed those unneeded lines from `.storage/core.device_registry` of the home-assistent data, which gave back the missing performance.

**Comment by qguv** [comment]
I get an additional 10 seconds loading time when opening the app with multiple servers set up. Removing all but one server solves the issue for me.

**Comment by Saduff** [comment]
I used to have this issue even with only one server. But the issue somehow got fixed for me a long time ago. Don't remember when exactly, but up to a year ago.

**Comment by felipecrs** [comment]
At this point this is a bit anecdotal, but:

1. The issue isn't as bad as it once was
2. Sometimes the app still takes way too long to load, and when that happens, I give up in favor of the PWA

**Comment by PapaGei7** [comment]
> yes, this started for me somewhat recently too, although I can't pin point when. I keep HA up to date along with add ons and HACs, and I keep the android app up to date. I tried the clear cache in the android app work around which seemed to work, It slowed down within a few hours again though, so I tried another method I found (maybe on Reddit) that suggested to delete the known_devices.yaml file and restart HA - since then (now going on approximately 16hrs) its been good. waiting for the slow down to return though.
> 
> edit: found the source I mentioned above: https://www.reddit.com/r/homeassistant/comments/172umia/comment/k446m8t/?utm_source=share&utm_medium=web2x&context=3

Great, it took me a few weeks of repeated cache clearing, android restarts, analyzing android webview with AI help. I installed newer and older version of companion app, until I found your hint and that's it. I also played around with BLE trilateration and my known_devices.yaml had more than 60k lines !
Thank you very much.
